document/core/exec/instructions.rst

```diff
@@ -1118,7 +1276,81 @@ Reference Instructions
 :math:`\ARRAYINITDATA~x~y`
 ..........................
 
-.. todo:: Prose
+1. Assert: due to :ref:`validation <valid-array.init_data>`, the :ref:`defined type <syntax-deftype>` :math:`F.\AMODULE.\MITYPES[x]` exists.
+
+2. Let :math:`\deftype` be the :ref:`defined type <syntax-deftype>` :math:`F.\AMODULE.\MITYPES[x]`.
+
+3. Assert: due to :ref:`validation <valid-array.init_data>`, :math:`\deftype` is an :ref:`array type <syntax-arraytype>`.
+
+4. Let :math:`\TARRAY~\X{ft}` be the :ref:`array type <syntax-arraytype>` :math:`\deftype`. (todo: unroll)
```

Comment:
This todo is elsewhere resolved in #406, you can copy from there. (Which also tweaks the wording of the previous bullet.)